pkg/envoy/sds/response.go

```diff
@@ -78,23 +69,23 @@ func (s *sdsImpl) getSDSSecrets(cert *certificate.Certificate, requestedCerts []
 					Str("proxy", proxy.String()).Msgf("Error getting service cert %s for proxy", requestedCertificate)
 				continue
 			}
-			certs = append(certs, envoySecret)
+			sdsSecrets = append(sdsSecrets, envoySecret)
 
 		// A root certificate used to validate a service certificate is requested
 		case secrets.RootCertTypeForMTLSInbound, secrets.RootCertTypeForMTLSOutbound:
-			envoySecret, err := s.getRootCert(cert, *sdsCert)
+			envoySecret, err := getRootCert(cert, *sdsCert, meshCatalog, trustDomain)
 			if err != nil {
 				log.Error().Err(err).Str("proxy", proxy.String()).Msgf("Error getting root cert %s for proxy", requestedCertificate)
 				continue
 			}
-			certs = append(certs, envoySecret)
+			sdsSecrets = append(sdsSecrets, envoySecret)
```

Comment:
I think this logic of building the secrets should be builder.go, otherwise with the current implementation technically response.go is still handling some of the building logic for the secrets config. 

So getSDSSecrets and getRootCert would be methods in builder.go. The builder struct would have cert, requestedCerts, proxy, meshcatalog.ListServiceIdentitiesForService (we don't want to pass in the entire meshcatalog object, just what's necessary), and trustDomain as fields, and we set those when we create the builder object. 

Then, builder.Build() will call the getSDSSecrets and getRootCert methods and return the sdsResources.